signac/contrib/project.py

```diff
@@ -1565,25 +1555,15 @@ def init_project(cls, *args, root=None, workspace=None, make_dir=True, **kwargs)
                 )
         except LookupError:
             fn_config = _get_project_config_fn(root)
-            if make_dir:
-                _mkdir_p(os.path.dirname(fn_config))
+            _mkdir_p(os.path.dirname(fn_config))
             config = read_config_file(fn_config)
-            if workspace is not None:
-                config["workspace_dir"] = workspace
             config["schema_version"] = SCHEMA_VERSION
             config.write()
             project = cls.get_project(root=root)
             if name is not None:
                 project.doc[name_key] = name
             return project
```

Comment:
Both the `except` and the `else` have a `return project` at the end. I would move that return out of the `try/except/else` and remove the `else`.